frontend: single dropdown styling and refactoring

shonsirsha opened the pull request:
Refactored and re-styled the **single dropdown** component for the newly redesigned / revamped settings. This is so that we can re-use and already are re-using the this **single dropdown** component in this PR. 

We'll do the same for multi dropdown component (which is used for selecting active currencies) **in another PR.**

For detailed changes, please refer to the commit history.

Preview (light):
<img width="1482" alt="Screenshot 2023-05-01 at 17 44 48" src="https://user-images.githubusercontent.com/28676406/235481514-3b220f40-fa44-482b-b7c2-ca9661a9acc0.png">

Preview (dark):
<img width="1482" alt="Screenshot 2023-05-01 at 17 44 42" src="https://user-images.githubusercontent.com/28676406/235481531-219af338-df0c-45fc-a367-b334bd310ff2.png">

thisconnect commented:
nice code LGTM, please squash into 1 commit as this is kind of 1 change to the UI.

ui question: should the border of the dropdown be white in darkmode? cc @jadzeidan 

dropdown on send view has no border, just the icon:

![Screen Shot 2023-05-03 at 11 46 15](https://user-images.githubusercontent.com/546900/235883955-6a7bf541-6480-43c1-922e-402d76ed1ca3.png)

jadzeidan commented:
> nice code LGTM, please squash into 1 commit as this is kind of 1 change to the UI and rebase.

> ui question: should the border of the dropdown be white in darkmode? cc @jadzeidan

> dropdown on send view has no border, just the icon:

@thisconnect the dropdown on the send view is clear, because it is a different color to the background, so it doesn't need a boarder.

On the settings page, I think it makes sense to add a border to the dropdown but using the color `color-dark` (which would be var `--background`)

shonsirsha commented:
> nice code LGTM, please squash into 1 commit as this is kind of 1 change to the UI and rebase.


Rebased, change(s) made in regards to border, and squashed into 1 commit.  PTAL 🙏

jadzeidan commented:
> Thanks both of you!
> 
> @jadzeidan , this is how it looks like using `--background` on darkmode. Do you think it looks ok?
> 
> <img alt="Screenshot 2023-05-03 at 17 00 14" width="322" src="https://user-images.githubusercontent.com/28676406/235956485-7bbb59fd-d684-44e4-9828-908233e6d188.png">

Thanks, looks good to me. 

Maybe would be better if the border was the same color as the chevron. But changing the color variable would mess up the light mode version since light mode uses `--background` too (which looks good on light mode), so we would need to introduce a new variable if want to change the boarder color and keep the code clean. But for now I think the dark border looks fine.

jadzeidan commented:
@shonsirsha yeah that looks better. So let's use that color if you are adding a new variable anyways. :)

shonsirsha commented:
> @shonsirsha yeah that looks better. So let's use that color if you are adding a new variable anyways. :)

Great!  @thisconnect feel free to review (again? 😉 )

thisconnect commented on frontends/web/src/routes/new-settings/components/singledropdown/singledropdown.module.css:
```diff
@@ -0,0 +1,50 @@
+}
+
+.valueContainer {
+    font-size: 16px;
```
isn't there a CSS variable that you can use for the font-size here?

shonsirsha replied:
Actually, we do. It's

```css
   /* wizard */
    --size-wizard-text: 16px;
```

sounds a bit specific (the naming) so I refrain from using it. Should we use this?

jadzeidan replied:
I think 14px is good. That's the same size we have for the current currencies.

thisconnect replied:
Looking at variables.css I think we should cleanup this up globally, not in this PR.

14px is ok, normal text input fields seem to also use 14px.

shonsirsha replied:
> Looking at variables.css I think we should cleanup this up globally, not in this PR.

Yeah, not going to modify already existing variables here thus going with `--size-default`  now (14px).  Had the impression of the text in the design being 16px not 14px..  but clearly 14px it is:)

thisconnect replied:
> Had the impression of the text in the design being 16px not 14px.. but clearly 14px it is:)

yeah me too, I think in view.modules.css it is bigger (18px). Generally we should move to bigger font one day (soonish).

thisconnect commented on frontends/web/src/routes/new-settings/components/singledropdown/singledropdown.tsx:
```diff
@@ -0,0 +1,66 @@
+      options={options}
+    />
+  );
+};
```
there is many `any` here, I haven't really looked into but could you check if it is possible to type it?

shonsirsha replied:
I'll see if we could use generics.

thisconnect replied:
it would be nice if it is more or less straight forward to do. else we can also merge as is.

thisconnect commented on frontends/web/src/routes/new-settings/components/singledropdown/singledropdown.module.css:
```diff
@@ -0,0 +1,50 @@
+.dropdown {
+    appearance: none;
+    background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='24' height='24' fill='none' stroke='currentColor' stroke-linecap='round' stroke-linejoin='round' stroke-width='2' aria-labelledby='chevron down' color='%23777'%3E%3Cdefs/%3E%3Cpolyline points='6 10 12 16 18 10'/%3E%3C/svg%3E%0A");
```
`currentColor` only works if the SVG elements are in the DOM (afaik), but not over CSS. Could you hardcode the color?

shonsirsha replied:
It does work it seems like.. Without current colour it'll be not visible. Also tried changing it to red and the chevron changed to red, too:)

thisconnect replied:
yes red is a color, but currentColor is like CSS inherit, but that doens't work if the svg is in backrgound-imag data url, only if it is in the dom.

shonsirsha replied:
Sorry, I understood what you meant. I was thinking the `stroke` attribute. But im not sure what you meant either by "doesn't work" here because it definitely inherits something, as leaving it blank will make the chevron invisible.
<img width="343" alt="Screenshot 2023-05-08 at 12 55 05" src="https://user-images.githubusercontent.com/28676406/236806576-4e4597a7-131a-4b04-82b8-c8af66f21d71.png">

Anyway, doesn't really matter i guess. But I went with your suggestion w/ hardcoding it with `#7777` to be safe. :)

thisconnect replied:
gray must be some kind of fallback, I made a quick and dirty test, inline svg vs. css background with svg
both use currentColor, the parent defines green color. the inline svg inherits green. the css background doesn't, as it can't I think as it doesn't know about the context. same with external svg. 

https://jsfiddle.net/2scndvbe/

shonsirsha replied:
Oh haha, such a neat coincidence that it fallbacks to the exact gray we're using too! :) Thanks Eni 👍

shonsirsha commented:
@thisconnect  Could you give another look here? Thanks! 🙏

thisconnect commented on frontends/web/src/routes/new-settings/components/appearance/defaultCurrencyDropdownSetting.tsx:
```diff
@@ -1,32 +1,7 @@
-  }
-  options={options}
-/>;
+import { store } from '../../../../components/rates/rates';
```
nit:  use  same import as setActiveFiat

thisconnect reviewed (approved):
👍  untested LGTM, with small nit
